tests/DistIL.Tests/IR/MatchingTests.cs

```diff
@@ -0,0 +1,135 @@
+﻿namespace DistIL.Tests.IR;
+
+using DistIL.AsmIO;
+using DistIL.IR;
+using DistIL.IR.Utils;
+
+[Collection("ModuleResolver")]
+public class MatchingTests
+{
+    private readonly ModuleResolver _modResolver;
+    private MethodDesc _stub;
+    private readonly ModuleDef _module;
+    private readonly TypeDef _testType;
+
+
+    public MatchingTests(ModuleResolverFixture mrf)
+    {
+        _modResolver = mrf.Resolver;
+
+        _module = _modResolver.Create("Test");
+        _testType = _modResolver.Resolve("TestAsm").FindType(null, "Stub")!;
+        _stub = _testType.FindMethod("StubMethod");
```

Comment:
This is still crashing for me, I got it working with the following but some of the tests are still failing

```suggestion
        _module = _modResolver.Resolve("TestAsm");
        _testType = _module.FindType(null, "MatcherStub")!;
        _stub = _testType.FindMethod("StubMethod");
```

[xUnit.net 00:00:01.06]     DistIL.Tests.IR.MatchingTests.TestNumberOperator [FAIL]
[xUnit.net 00:00:01.06]     DistIL.Tests.IR.MatchingTests.TestCallMatchOutput [FAIL]
[xUnit.net 00:00:01.06]     DistIL.Tests.IR.MatchingTests.TestUnary [FAIL]
[xUnit.net 00:00:01.07]     DistIL.Tests.IR.MatchingTests.Test_Strings [FAIL]